client/components/sites-dropdown/test/index.js

```diff
@@ -23,104 +24,130 @@ describe( 'index', function() {
 		mockery.registerSubstitute( 'matches-selector', 'component-matches-selector' );
 	} );
 
-	let SitesDropdown;
+	describe( 'Connected(SitesDropdown)', function() {
+		let ConnectedSitesDropdown;
 
-	before( function() {
-		SitesDropdown = require( '../index.jsx' );
-	} );
-
-	describe( 'component rendering', function() {
-		it( 'should render a dropdown component initially closed', function() {
-			const sitesDropdown = shallow( <SitesDropdown /> );
-			expect( sitesDropdown.hasClass( 'sites-dropdown' ) ).to.be.true;
-			expect( sitesDropdown.hasClass( 'is-open' ) ).to.be.false;
+		before( function() {
+			ConnectedSitesDropdown = require( '../index.jsx' ).default;
```

Comment:
We have never tested Component wrapped with `connect` so far. I'm not opposed to this approach. From what I see in this example you get additional verification for your connect method, which can be useful.

We could also avoid using `redux-mock-store` by testing a plain component with all props injected when necessary. At the same time we could also export `mapStateToProps` and test it independently.

My personal preference is the latter option, because it makes testing setup much easier.